Refactor of codebase

claw726 opened the pull request:
- Refactored much of the code base to be component-based, instead of relying on the same code across multiple files.
- Changed much of the CSS to match EVE Frontier's dark theme.
- Added some type checking when receiving data from the websocket server.

claw726 committed:
Refactored code to be reused across multiple pages

Total redo of UI

claw726 committed:
Added new image

claw726 committed:
delete screenshot

kandrsn99 commented:
The language selector is 'funky.' Basically, everything here is ready to be merged. But when a new language is selected, I have to refresh the page. This error only appears when I have an incorrect search, resulting in an error.

claw726 committed:
FIX: Language update error during incorrect search

claw726 commented:
Fixed the problem. The function was returning text, not inline HTML. Fixed it so the text should dynamically update

claw726 committed:
FIX: CSS to make text more readable

Commit:
FEAT:
 - Revamped kill-mail card design
 - Added dynamic timezone conversion (and option to keep it UTZ)

Commit:
FIX: Dynamic Resizing of cards and container.

claw726 commented:
Did some more work with CSS and timezones.

claw726 committed:
FIX: added node modules and envs to gitignore

claw726 committed:
CHORE: Cleaned up utils.js

claw726 committed:
FIX: Fixed localization errors for incident cards

claw726 committed:
FEAT:
- Added placeholder images for future profile photos
- restyled kill cards to be list items

claw726 committed:
FEAT:
- clicking on player names or systems redirects user to search result of the respective object.
- Added more stats to player and system search results

BROKEN:
- need to add CSS for these stat views.

claw726 committed:
CHORE: Refactor CSS Files

claw726 committed:
FIX: Timezones not changing when toggling TZ

claw726 committed:
CHORE: Removed System ID from kill cards

claw726 committed:
CHORE: Added fonts to prefetch URLs

claw726 committed:
FEAT: Added landing content

TODO: Add translations for landing content

claw726 committed:
Added translations to hero section

claw726 committed:
FIX: Added translations to hero content and TZ toggle

claw726 committed:
CHORE: Removed curved edges and aligned CSS in kill cards

Commit:
FIX: Navbar remains in view at all times

Commit:
FIX: buttons now extend down when hovered

Commit:
CHORE: Added types to API docs

Commit:
FEAT: Addred redirection in stat tables

Commit:
FIX: CSS for min-width on name

Commit:
CHORE: lint job

Commit:
CHORE: Fixed translation alignment

Commit:
CHORE: Moved translations to separate file to maintainability

Commit:
CHORE: Added comment on where translations come from

Commit:
FIX: Added TZ toggle to search page

Commit:
CHORE: Linted all files

Commit:
Removed unused CSS files.

Commit:
Added <main> tags to each HTML doc

Commit:
Added <aria-live> tag to indicate content may automatically update

Commit:
FEAT: Added caching for metric data.

claw726 committed:
FEAT: Added timeranges to metric views and created individual kill mail pages.

claw726 committed:
FEAT: New background

claw726 committed:
CHORE: Migrated project to Vite

claw726 committed:
CHORE: Updated README, outlining how to bundle the project with Vite

claw726 committed:
FEAT: Added an individual killmail page

claw726 committed:
FIX: Translations now work on killmail page

kandrsn99 commented:
You have been added as a maintainer. Check your notifications. This should allow you to merge requests yourself from now.

claw726 committed:
CHORE: Removed /dist/ from .gitignore

claw726 committed:
FEAT: Enhanced SEO and performance with meta tags, lazy loading, and new sitemap

claw726 committed:
CHORE: Added missing translations

claw726 committed:
CHORE: Added @biomejs/biome as a dev dependency and updated scripts for formatting, linting, and checking

Commit:
FEAT: Introduced new API endpoints for health checks, incidents, and locations; updated type definitions for incidents and locations.

Commit:
Merge branch 'main' of github.com:claw726/Alpha-Strike-Main

Commit:
REFAC: Removed 'export' keyword from API function declarations for consistency and improved readability.

Commit:
STYLE: Refactored navigation CSS for improved readability and responsiveness; adjusted styles for mobile and desktop views, including logo and hamburger menu adjustments.

claw726 committed:
CHORE: Updated configuration files and added husky with lint-staged for pre-commit hooks

claw726 committed:
Merge branch 'main' of github.com:claw726/Alpha-Strike-Main

claw726 committed:
REFAC: Updated asset references in HTML files and removed obsolete JavaScript and CSS files to streamline the project structure and improve loading performance.

Commit:
DOCS: Updated README.md for improved clarity and organization; added folder structure details, prerequisites, and installation instructions.

Commit:
STYLE: Enhanced CSS formatting for improved readability; added compact media queries for better responsiveness on medium and small screens.

Commit:
CICD: Regenerated DIST

Commit:
FEAT: Implemented navbar search functionality with dropdown; added search input and button, integrated search logic, and enhanced CSS for improved styling and responsiveness.

Commit:
CICD: regenerated DIST

Commit:
REFAC: Refactored HTML structure for improved readability and consistency; updated JavaScript imports for incident card functionality and adjusted CSS for enhanced styling and responsiveness.

Commit:
CICD: Rebuilt Dist
